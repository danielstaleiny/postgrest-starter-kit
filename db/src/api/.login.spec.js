const ROOT = process.cwd()
const resetdb = require(`${ROOT}/util/reset-db.js`)
const { get, post, put, patch, del } = require(`${ROOT}/util/fetch.js`)(
  'http://localhost:3000/'
)

describe('/rpc/login', () => {
  beforeAll(() => {
    return resetdb()
  })
  afterAll(() => {
    return resetdb()
  })

  test('success 200', async () => {
    const res = await post('rpc/login')({
      body: {
        email: 'alice@email.com',
        password: 'pass',
      },
    })
    res.body.csrf = 'checked'
    res.body.token = 'checked'
    res.body.refresh_token = 'checked'
    expect(res).toMatchSnapshot()
  })

  test('fail 404', async () => {
    const res = await post('rpc/login')({
      body: {
        email: 'random@email.com',
        password: 'nonexistent',
      },
    })
    expect(res).toMatchSnapshot()
  })
})
