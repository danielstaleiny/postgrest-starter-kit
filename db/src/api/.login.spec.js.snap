// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/rpc/login fail 404 1`] = `
Object {
  "body": Object {
    "code": "28P01",
    "details": "Invalid email or password",
    "hint": "Make sure your email and password are correct!",
    "message": "invalid_password",
  },
  "status": 403,
  "statusText": "Forbidden",
}
`;

exports[`/rpc/login success 200 1`] = `
Object {
  "body": Object {
    "csrf": "checked",
    "email": "alice@email.com",
    "id": 1,
    "name": "alice",
    "refresh_token": "checked",
    "role": "webuser",
    "token": "checked",
  },
  "status": 200,
  "statusText": "OK",
}
`;
